### PR TITLE
Improve accessibility (part 3)

### PR DESCRIPTION
**Pull request by sorin-davidoi**
Follow-up to #4369 and #4377. Mostly fixes errors discovered by WAVE Web Accessibility Tool.

cc: @yiskah

### CODE REVIEW

**Comment by sorin-davidoi** [comment]
I think the warnings are wrong? `input` is inside `label` (which is what WAVE Web Accessibility Tool suggests).

**Comment by unarist** [comment]
Looks like it was fixed last month? https://github.com/evcohen/eslint-plugin-jsx-a11y/releases/tag/v6.0.0

File: app/javascript/mastodon/components/column_header.js [hunk 1]
```diff
@@ -133,15 +135,15 @@ export default class ColumnHeader extends React.PureComponent {
 
     return (
       <div className={wrapperClassName}>
-        <div role='heading' tabIndex={focusable && '0'} className={buttonClassName} aria-label={title} onClick={this.handleTitleClick}>
+        <h1 role='heading' tabIndex={focusable && '0'} className={buttonClassName} aria-label={title} onClick={this.handleTitleClick}>
```
Comment (unarist) [inline app/javascript/mastodon/components/column_header.js]: It's heading element now, so `role='heading'` can be removable.

**Comment by Gargron** [comment]
>Form controls using a label to identify them must be programmatically associated with the control using htmlFor

Hwhat?

**Comment by sorin-davidoi** [comment]
Should I just disable the errors?

I've added the `inert` attribute to the siblings of `modal-root` when it is opened. I'm deliberately not including [the polyfill](https://github.com/WICG/inert#notes-on-the-polyfill) because I didn't test it's performance (relies on mutation observers).

**Review by Gargron — APPROVED** [review]

**Comment by yiskah** [comment]
<3 you're the best sorin

**Pull request merged**